src/main/java/jfxtras/labs/internal/scene/control/skin/LedBargraphSkin.java

```diff
@@ -137,12 +137,16 @@ public void changed(ObservableValue<? extends Number> ov, Number oldValue, Numbe
                 int currentLedPeakIndex = 0;
                 for (int i = 0 ; i < control.getNoOfLeds() ; i++) {
                     if (Double.compare(i * stepSize.doubleValue(), newValue.doubleValue()) <= 0) {
-                        ledList.get(i).setOn(true);
+                        if (newValue.doubleValue() == 0.0) {
+                            ledList.get(i).setOn(false);
+                        } else {
+                            ledList.get(i).setOn(true);
+                        }
                         currentLedPeakIndex = i;
                     } else {
                         ledList.get(i).setOn(false);
                     }
-                    ledList.get(peakLedIndex).setOn(true);
+                    if (control.isPeakValueVisible()) ledList.get(peakLedIndex).setOn(true);
```

Comment:
For readability, I would recommend putting `ledList.get(peakLedIndex).setOn(true);` on the next line and with curly-brackets. It would also conform with the existing style of the code.